libnavigation-core/src/main/java/com/mapbox/navigation/NavigationController.kt

```diff
@@ -51,27 +61,49 @@ class NavigationController(
     private fun paramsProvider(type: MapboxNavigationModuleType): Array<Pair<Class<*>?, Any?>> {
         return when (type) {
             HybridRouter -> arrayOf(
-                Pair(Router::class.java, NavigationModuleProvider.createModule(OnboardRouter, ::paramsProvider)),
-                Pair(Router::class.java, NavigationModuleProvider.createModule(OffboardRouter, ::paramsProvider))
+                Router::class.java to NavigationModuleProvider.createModule(
+                    OnboardRouter,
+                    ::paramsProvider
+                ),
+                Router::class.java to NavigationModuleProvider.createModule(
+                    OffboardRouter,
+                    ::paramsProvider
+                )
+            )
+            OffboardRouter -> arrayOf(
+                Context::class.java to application,
+                String::class.java to mapboxToken
             )
-            OffboardRouter -> arrayOf()
             OnboardRouter -> arrayOf(
-                Pair(MapboxNativeNavigator::class.java, navigator)
+                MapboxNativeNavigator::class.java to navigator
             )
             DirectionsSessionModule -> arrayOf(
-                Pair(Router::class.java, NavigationModuleProvider.createModule(HybridRouter, ::paramsProvider))
+                Router::class.java to NavigationModuleProvider.createModule(
+                    HybridRouter,
+                    ::paramsProvider
+                ),
+                Point::class.java to origin,
+                rawType<List<Point>>() to waypoints,
```

Comment:
Generic types are erased in the runtime anyway, so a simple `List::class.java to waypoint` is enough here.

Reply:
yep, in bytecode it's just `List`. would like to show that it's `List<Point>`type explicitly.
replaced